sdk/core/azure-core/inc/uuid.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <new> // for placement new
+#include <string>
+#include <utility> // for swap and move
+#include <random>
+
+namespace Azure { namespace Core {
+
+  class UUID {
+
+  private:
+    static const int UUIDSize = 16;
+
+    uint8_t m_uuid[UUIDSize];
+    // The UUID reserved variants.
+    static constexpr uint8_t ReservedNCS = 0x80;
+    static constexpr uint8_t ReservedRFC4122 = 0x40;
+    static constexpr uint8_t ReservedMicrosoft = 0x20;
+    static constexpr uint8_t ReservedFuture = 0x00;
+
+  public:
+    UUID()
```

Comment:
I still think that it would be better if it was a static method `Uuid::RandomUuid()` that produces a random UUID, and not the default constructor. Think about having this class as a member, and then default-initializing.

Reply:
You can even make the default constructor private now, if you don't want to worry about the default implementation, so that with private default constructor the only way instances of this class can be produced now would be either from `Uuid::RandomUuid()`, or when they are assigned from/to by using default auto-generated copy constructor and assignment operator.

Reply:
I agree with @antkmsft on the static function. But I don't think the function name should contain `random`.

Reply:
OK, let's do the following:
The ctor should create a zero-UUID (all bytes 0).

A static New() function should return a new UUID instance with a value. NOTE: If pseudo-random generator is used, then it needs to be seeded. We don't want the same UUID values to be generated again if the process is stopped and restarted.

We need a method that returns the UUID as a string (hex digits [a-f] in lowercase).

== and != operators should be overloaded so that comparisons can be done.

It would be nice to have a static Parse function or perhaps a ctor that takes a string and parses it back to a UUID instance. This is low priority.

As or casing of the class, we can review with Casey/Billy: Is it common in C++ to keep acronyms in all uppercase or in mixed case. I know that in .NET, the rule is that acronyms should be mixed case (like Http, Guid) unless it is a 2-letter acronym (like IO). Differnet languages do this different ways and we should be idiomatic for C++ here.

Reply:
I hate quoting myself, but this discussion is split into several comment sections.
> Well, storage service returns `"InvalidHttpVerb"` or `"UnsupportedHttpVerb"` in response error reason phrase, while swagger defines `Content-MD5` and `blob-HTTP-headers` and `GetRangeContentCRC64` so it is also not consistent there.
> However, for swagger generator's sake, I would vote for acronyms to be written in all caps. @ahsonkhan for awareness.